modules/fs_extensions.js

```diff
@@ -0,0 +1,87 @@
+const fsPromises = require("fs").promises;
+const log = require("electron-log");
+
+"use strict";
+
+const extensions = function() {
+    /**
+     * catches NotFound-errors 
+     * 
+     * used to check files/pathes and handle NotFound-errors as 'does not exist'
+     * @param {BoolAction} func function to execute
+     * @returns {Promise<boolean>} false if NotFound-error occoured, otherwise result from func will be returned
+     */
+    async function Exists(func) {
+        try {
+            return await func();
+        } catch (e) {
+            if (!IsNotFoundError(e))
+                throw e;
+            return false;
+        }
+    }
+
+    function IsNotFoundError(e) {
+        return e.code === "ENOENT";
+    }
+
+
+    return {
+        /**
+         * moves a file to a new location
+         * returns if oldPath is not a file
+         * creates directories if needed
+         * @param {string} oldPath origin path
+         * @param {string} newPath destination path
+         */
+        async move(oldPath, newPath) {
+            if (!await this.fileExists(oldPath))
+                return;
+        
+            let newDirectory = global.path.dirname(newPath);
+            await this.ensureDirectoryExists(newDirectory);
+        
+            await fsPromises.rename(oldPath, newPath);
+        },
+
+        /**
+         * creates directory, if it does not exist
+         * does nothing otherwise
+         * @param {string} directory directory to check
+         * @returns {Promise<boolean>} true if directory was created, otherwise false
+         */
+        async ensureDirectoryExists(directory) {
+            if(!await this.pathExists(directory)){
+                await fsPromises.mkdir(directory, {recursive: true});
+                log.log("Made the directory: " + directory);
+                return true;
+            }
+
+            return false;
+        },
```

Comment:
This fn pretty much has the same effect as running
```js
await fsPromises.mkdir(directory, { recursive: true });
```
If the directory exists, `mkdir` does nothing, and reports no error (this only happens when `recursive: true`)

Reply:
This is interesting to know :D
I tried to keep the code as close to the existing (the old code checks the directory before creating it)
there is also some rework going on on an other branch iirc
so i don't think we want to change the code here ... at least not rn ^^
@rob5300 any thoughts on how/where/when to implement this?